openhcl/diag_client/src/lib.rs

```diff
@@ -574,28 +625,19 @@ impl DiagClient {
     ///
     /// This can be used to support extension RPCs that are not part of the main
     /// diagnostics service.
-    pub fn custom_call<F, R, T, U>(
-        &self,
-        rpc: F,
-        input: T,
-        timeout: Option<Duration>,
-    ) -> mesh::OneshotReceiver<Result<U, Status>>
-    where
-        F: FnOnce(T, mesh::OneshotSender<Result<U, Status>>) -> R,
-        R: mesh_rpc::service::ServiceRpc,
-        U: mesh::MeshPayload,
-    {
-        self.ttrpc.start_call(rpc, input, timeout)
+    pub fn custom_call(&self) -> mesh_rpc::client::CallBuilder<'_> {
+        self.ttrpc.call()
     }
 
     /// Crashes the VM.
     pub async fn crash(&self, pid: i32) -> anyhow::Result<()> {
         self.ttrpc
-            .call(
+            .call()
+            .start(
```

Comment:
maybe worth adding a `simple_call` helper?

Reply:
I prefer the builder pattern, because it hopefully leads people to think, "what else can I do with this"? And then maybe they'll realize they should set a deadline.

Reply:
This is reasonable, and in-line with my usual line of thinking when designing these sorts of things. sounds good.

Reply:
(OTOH, in the future I may end up requiring that you pass a `CancelContext`, which would make that even more obvious. At that point, a `simple_call` might make sense. The main reason I hesitate to do this is that TTRPC does not support non-timeout-based cancellation (whereas gRPC does), so this could be misleading.)